hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -4738,6 +4738,12 @@ public long getMissingReplOneBlocksCount() {
     // not locking
     return blockManager.getMissingReplOneBlocksCount();
   }
+
+  @Metric({"BadlyDistBlocks", "Number of Badly Distributed Blocks"})
+  public long getBadlyDistributedBlocksCount() {
+    // not locking
+    return blockManager.getBadlyDistributedBlocksCount();
+  }
```

Comment:
This metric looks like a duplicate of below. Is that intentional?

Reply:
I think both are needed since the BlockManager gets the count, BadlyDistBlocks exposes metric for Metrics2 and theres also BadlyDistributedBlocks  metric for MBean

Reply:
Also getBadlyDistributedBlocksCount is not locking while getBadlyDistributedBlocks isnt